lib/src/component_declaration/flux_component.dart

```diff
@@ -228,24 +229,40 @@ abstract class FluxUiStatefulComponent2<TProps extends FluxUiProps, TState exten
 ///
 /// Private so it will only get used in this file, since having lifecycle methods in a mixin is risky.
 mixin _FluxComponentMixin<TProps extends FluxUiProps> on component_base.UiComponent<TProps>, BatchedRedraws {
-  static final Logger _logger = Logger('over_react._FluxComponentMixin');
-
   /// List of store subscriptions created when the component mounts.
   ///
   /// These subscriptions are canceled when the component is unmounted.
   List<StreamSubscription> _subscriptions = [];
 
   void _validateStoreDisposalState(Store store) {
-    String message = 'Cannot listen to a disposed/disposing Store.';
+    if (store.isOrWillBeDisposed) {
+      final componentName = getDebugNameForDartComponent(this);
+
+      // Include the component name in the logger name so that:
+      // 1. it's included in the log somewhere
+      // 2. logs from the same component can be easily grouped together
+      final logger = Logger('over_react._FluxComponentMixin.$componentName');
+
+      var storeNameOrType = store.disposableTypeName;
+      // Detect if they don't override disposableTypeName by checking for the default name.
+      if (storeNameOrType == 'Store') {
+        storeNameOrType = store.runtimeType.toString();
+      }
 
-    var isDisposedOrDisposing = store.isOrWillBeDisposed ?? false;
+      final message = 'Cannot listen to a disposed/disposing Store.'
+          ' (storeNameOrType: $storeNameOrType, shouldBatchRedraw: $shouldBatchRedraw)';
```

Comment:
Would it be better to use a `ContextualMessage` and put `storeNameOrType` and `shouldBatchRedraw` in the context of the message instead?

Reply:
I really wanted to do that, but unfortunately I'd have to depend on app_intelligence_dart to get that class (which we can't do since this is an OSS repo, but even if it weren't I'd be wary about).

It'd be nice if you could pass in a `Map` or some other core type and have it treat it the same as a `ContextualMessage`.

Reply:
Ahh. I didn't realize that was an AID-specific class.